packages/bvaughn-architecture-demo/src/suspense/createGenericCache.ts

```diff
@@ -1,12 +1,38 @@
+import { handleError } from "protocol/utils";
 import { Record, STATUS_PENDING, STATUS_REJECTED, STATUS_RESOLVED, Wakeable } from "./types";
 import { createWakeable } from "../utils/suspense";
 
+export interface GenericCache<TParams extends Array<any>, TValue> {
+  getValueSuspense(...args: TParams): TValue;
+  getValueAsync(...args: TParams): Promise<TValue>;
+  getValueIfCached(...args: TParams): TValue | undefined;
```

Comment:
nit: I think methods like this should return `null` for uncached objects. It's more explicit. (`undefined` can indicate an unhandled case since many collection types return that value for missing keys).

Reply:
Personally I prefer to avoid `null` whenever possible because I don't like the code being littered with `null | undefined`. Plus this is essentially a `Map` lookup so returning `undefined` seems consistent to me.

Reply:
Why would code be `null | undefined` ? A `null` return type is explicit. You only need to check if null or not.